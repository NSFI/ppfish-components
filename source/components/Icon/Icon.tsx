import * as React from 'react';
import classNames from 'classnames';
import omit from 'omit.js';
import './Icon.less'


export interface IconProps {
  type: string;
  className?: string;
  title?: string;
  onClick?: React.MouseEventHandler<any>;
  spin?: boolean;
  style?: React.CSSProperties;
}

const Icon = (props: IconProps) => {
  const { type, className = '', spin } = props;
  const classString = classNames({
    fishdicon: true,
    'fishdicon-spin': !!spin || type === 'loading',
    [`fishdicon-${type}`]: true,
  }, className);
  return <i {...omit(props, ['type', 'spin'])} className={classString} />;
};

export default Icon;
