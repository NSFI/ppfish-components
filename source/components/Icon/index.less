@import "../../assets/css/lib/ant-theme-vars";
