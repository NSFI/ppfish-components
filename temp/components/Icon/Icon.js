var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import * as React from 'react';
import classNames from 'classnames';
import omit from 'omit.js';
var Icon = function (props) {
    var _a;
    var type = props.type, _b = props.className, className = _b === void 0 ? '' : _b, spinning = props.spinning;
    var classString = classNames((_a = {
            fishdicon: true,
            'fishdicon-spin': spinning || type === 'loading'
        },
        _a["fishdicon-" + type] = true,
        _a), className);
    return React.createElement("i", __assign({}, omit(props, ['type', 'spinning']), { className: classString }));
};
export default Icon;
