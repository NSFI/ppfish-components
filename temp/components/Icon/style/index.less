@import (reference) "../../../assets/css/index";
