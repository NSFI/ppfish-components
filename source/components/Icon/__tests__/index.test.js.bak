import React from 'react';
import { render } from 'enzyme';
import Icon from '..';

describe('Icon', () => {
  it('should render to a <i class="xxx"></i>', () => {
    const wrapper = render(
      <Icon type="appstore" className="my-icon-classname" />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
