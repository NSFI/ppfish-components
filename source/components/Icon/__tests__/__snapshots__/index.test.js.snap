// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render to a <i class="xxx"></i> 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "fishdicon fishdicon-appstore my-icon-classname",
    },
    "children": Array [],
    "name": "i",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
